hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoCreateHandler.java

```diff
@@ -56,14 +84,203 @@ public void preHandle(@NonNull final PreHandleContext context) {
-     * change.
+     * This method is called during the handle workflow. It executes the {@code CryptoCreate}
+     * transaction, creating a new account with the given properties.
+     * If the transaction is successful, the account is created and the payer account is charged
+     * the transaction fee and the initial balance of new account and the balance of the
+     * new account is set to the initial balance.
+     * If the transaction is not successful, the account is not created and the payer account is
+     * charged the transaction fee.
      *
      * @throws NullPointerException if one of the arguments is {@code null}
+     * @throws HandleException      if the transaction is not successful due to payer
+     * account being deleted or has insufficient balance or the account is not created due to
+     * the usage of a price regime
+     */
+    public void handle(
+            @NonNull final HandleContext handleContext,
+            @NonNull final TransactionBody txnBody,
+            @NonNull final WritableAccountStore accountStore,
+            @NonNull final CryptoCreateRecordBuilder recordBuilder,
+            @NonNull final boolean areCreatableAccounts) {
+        final var op = txnBody.cryptoCreateAccount();
+
+        // validate fields in the transaction body that involves checking with
+        // dynamic properties or state
+        final ResponseCodeEnum validationResult = validateSemantics();
+        if (validationResult != OK) {
+            throw new HandleException(validationResult);
+        }
+
+        // If accounts can't be created, due to the usage of a price regime, throw an exception
+        if (!areCreatableAccounts) {
+            throw new HandleException(MAX_ENTITIES_IN_PRICE_REGIME_HAVE_BEEN_CREATED);
+        }
+
+        // validate payer account exists and has enough balance
+        final var optionalPayer = accountStore.getForModify(
+                txnBody.transactionIDOrElse(TransactionID.DEFAULT).accountIDOrElse(AccountID.DEFAULT));
+        if (optionalPayer.isEmpty()) {
+            throw new HandleException(INVALID_PAYER_ACCOUNT_ID);
+        }
+        final var payer = optionalPayer.get();
+        final long newPayerBalance = payer.tinybarBalance() - op.initialBalance();
+        validatePayer(payer, newPayerBalance);
+
+        // Change payer's balance to reflect the deduction of the initial balance for the new
+        // account
+        final var modifiedPayer =
+                payer.copyBuilder().tinybarBalance(newPayerBalance).build();
+        accountStore.put(modifiedPayer);
+
+        // Build the new account to be persisted based on the transaction body
+        final var accountCreated = buildAccount(op, handleContext);
+        accountStore.put(accountCreated);
+
+        // set newly created account number in the record builder
+        final var createdAccountNum = accountCreated.accountNumber();
+        recordBuilder.setCreatedAccount(createdAccountNum);
```

Comment:
This raises a bit of an interesting question about how record-building responsibilities are divided.

I.e., what makes it the handler's responsibility to record any newly created entities, but _not_ record the related hbar changes that go in the `transferList`? 🤔

Reply:
I think even transferList also should be under handler responsibility ?

Reply:
As you suggested will think how to add it to `commit()`